### PR TITLE
Add Codespace environment with minimal workflow and notebook for JHU APL

### PR DESCRIPTION
**Pull request by kabilar**
- Due to opaque dependency conflicts, `suite2p` has to be installed after `workflow-calcium-imaging`, so it is run in the `postStartCommand` command.

### CODE REVIEW

File: notebooks/tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dj.config[\"custom\"] = {\"database.prefix\": \"neuro_\",\n",
+    "                       \"imaging_root_data_dir\": \"/workspaces/workflow-calcium-imaging/example_data\"}"
```
Comment (guzman-raphael) [inline notebooks/tutorial.ipynb:64]:
```suggestion
    "dj.config[\"custom\"] = {\"database.prefix\": \"neuro_\",\n",
    "                       \"imaging_root_data_dir\": \"../example_data\"}"
```

File: notebooks/tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imaging.MotionCorrection.populate(**populate_settings)"
```
Comment (guzman-raphael) [inline notebooks/tutorial.ipynb:279]: Got an error on this step when I ran it:

```
KeyError                                  Traceback (most recent call last)
/tmp/ipykernel_3552/1818698745.py in <module>
----> 1 imaging.MotionCorrection.populate(**populate_settings)

~/.local/lib/python3.7/site-packages/datajoint/autopopulate.py in populate(self, suppress_errors, return_exception_objects, reserve_jobs, order, limit, max_calls, display_progress, processes, make_kwargs, *restrictions)
    239                 tqdm(keys, desc=self.__class__.__name__) if display_progress else keys
    240             ):
--> 241                 error = self._populate1(key, jobs, **populate_kwargs)
    242                 if error is not None:
    243                     error_list.append(error)

~/.local/lib/python3.7/site-packages/datajoint/autopopulate.py in _populate1(self, key, jobs, suppress_errors, return_exception_objects, make_kwargs)
    290                 self.__class__._allow_insert = True
    291                 try:
--> 292                     make(dict(key), **(make_kwargs or {}))
    293                 except (KeyboardInterrupt, SystemExit, Exception) as error:
    294                     try:

~/.local/lib/python3.7/site-packages/element_calcium_imaging/imaging.py in make(self, key)
    799                             "block_width": s2p.ops["block_size"][1],
    800                             "block_depth": 1,
--> 801                             "block_count_y": s2p.ops["nblocks"][0],
    802                             "block_count_x": s2p.ops["nblocks"][1],
    803                             "block_count_z": len(suite2p_dataset.planes),

KeyError: 'nblocks'
```

Ran it again with your original setting for `imaging_root_data_dir` and got:

```
FileNotFoundError                         Traceback (most recent call last)
/tmp/ipykernel_19561/1299329365.py in <module>
----> 1 imaging.Processing.populate(**populate_settings)

~/.local/lib/python3.7/site-packages/datajoint/autopopulate.py in populate(self, suppress_errors, return_exception_objects, reserve_jobs, order, limit, max_calls, display_progress, processes, make_kwargs, *restrictions)
    239                 tqdm(keys, desc=self.__class__.__name__) if display_progress else keys
    240             ):
--> 241                 error = self._populate1(key, jobs, **populate_kwargs)
    242                 if error is not None:
    243                     error_list.append(error)

~/.local/lib/python3.7/site-packages/datajoint/autopopulate.py in _populate1(self, key, jobs, suppress_errors, return_exception_objects, make_kwargs)
    290                 self.__class__._allow_insert = True
    291                 try:
--> 292                     make(dict(key), **(make_kwargs or {}))
    293                 except (KeyboardInterrupt, SystemExit, Exception) as error:
    294                     try:

~/.local/lib/python3.7/site-packages/element_calcium_imaging/imaging.py in make(self, key)
    386 
    387         if task_mode == "load":
--> 388             method, imaging_dataset = get_loader_result(key, ProcessingTask)
    389             if method == "suite2p":
    390                 if (scan.ScanInfo & key).fetch1("nrois") > 0:
...
---> 52                 "Suite2p output result files not found at {}".format(suite2p_dir)
     53             )
     54 

FileNotFoundError: Suite2p output result files not found at /workspaces/workflow-calcium-imaging/example_data/subject1/session1/suite2p
```

File: notebooks/tutorial.ipynb [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# from workflow_calcium_imaging.pipeline import *\n",
+    "# imaging.schema.drop()\n",
+    "# scan.schema.drop()\n",
+    "# session.schema.drop()\n",
+    "# subject.schema.drop()\n",
+    "# lab.schema.drop()"
```
Comment (guzman-raphael) [inline notebooks/tutorial.ipynb:395]: Was cleaning up to prepare for a new run and tried this but got an error as well. Likely the drop order just needs to be adjusted.

```
OperationalError                          Traceback (most recent call last)
/tmp/ipykernel_3552/3274683739.py in <module>
      1 from workflow_calcium_imaging.pipeline import *
----> 2 imaging.schema.drop()
      3 scan.schema.drop()
      4 session.schema.drop()
      5 subject.schema.drop()

~/.local/lib/python3.7/site-packages/datajoint/schemas.py in drop(self, force)
    362             try:
    363                 self.connection.query(
--> 364                     "DROP DATABASE `{database}`".format(database=self.database)
    365                 )
    366                 logger.debug(

~/.local/lib/python3.7/site-packages/datajoint/connection.py in query(self, query, args, as_dict, suppress_warnings, reconnect)
    338         cursor = self._conn.cursor(cursor=cursor_class)
    339         try:
--> 340             self._execute_query(cursor, query, args, suppress_warnings)
    341         except errors.LostConnectionError:
    342             if not reconnect:

~/.local/lib/python3.7/site-packages/datajoint/connection.py in _execute_query(cursor, query, args, suppress_warnings)
    294                 cursor.execute(query, args)
...
    141     if errorclass is None:
    142         errorclass = InternalError if errno < 1000 else OperationalError
--> 143     raise errorclass(errno, errval)

OperationalError: (3730, "Cannot drop table '__segmentation' referenced by a foreign key constraint '__scan_level_report_ibfk_1' on table '__scan_level_report'.")
```

Found better success using an approach like this:

```python
def drop_databases(databases):
    import pymysql.err
    conn = dj.conn()

    with dj.config(safemode=False):
        for database in databases:
            schema = dj.Schema(f'{dj.config["custom"]["database.prefix"]}{database}')
            while schema.list_tables():
                for table in schema.list_tables():
                    try:
                        conn.query(f"DROP TABLE `{schema.database}`.`{table}`")
                    except pymysql.err.OperationalError:
                        print(f"Can't drop `{schema.database}`.`{table}`. Retrying...")
            schema.drop()

drop_databases(databases=['imaging_report', 'imaging', 'scan', 'session', 'subject', 'lab', 'reference'])
```

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
Thanks @kabilar! :handshake: 

Tried running this workflow but ran into some issues. Also, providing some suggestions for further improvement.

File: requirements.txt [hunk 4]
```diff
@@ -1,13 +1,9 @@
...
+element-animal>=0.1.2
 element-session>=0.1.2
+element-interface>=0.5.0
+element-calcium-imaging>=0.3.0
 ipykernel>=6.0.1
 jupytext>=1.13.7
-nd2
-sbxreader @ git+https://github.com/datajoint-company/sbxreader
-scanreader @ git+https://github.com/atlab/scanreader.git
-suite2p @ git+https://github.com/datajoint-company/suite2p.git
```
[the remark this answers is not in the document]
Comment (kabilar) [reply requirements.txt:13]: There is a confusing dependency conflict.  I found that suite2p can be installed when added to the `postStartCommand`.  Not ideal but it works.

File: .devcontainer/Dockerfile [hunk 5]
```diff
@@ -1,11 +1,24 @@
-# Note: You can use any Debian/Ubuntu based image you want. 
 FROM mcr.microsoft.com/devcontainers/python:3.7-bullseye
 
+ENV GITHUB_USER datajoint
+ENV ELEMENT calcium-imaging
```
[the remark this answers is not in the document]
Comment (kabilar) [reply .devcontainer/Dockerfile]: Removed.  Was used in a previous version.

File: .devcontainer/devcontainer.json [hunk 6]
```diff
@@ -15,11 +15,11 @@
 		"ghcr.io/eitsupi/devcontainer-features/jq-likes:1": {},
 		"ghcr.io/guiyomh/features/vim:0": {}
 	},
-	// "onCreateCommand": "pip install -e .",
-	"postStartCommand": "mkdir -p example_data && s3fs djhub.vathes.datapub.elements:/workflow-calcium-imaging-data-v2 example_data -o multipart_size=530,endpoint=us-east-1,url=http://s3.amazonaws.com,public_bucket=1 && MYSQL_VER=8.0 docker compose down && MYSQL_VER=8.0 docker compose up --build --wait",
+	"onCreateCommand": "pip install -e .",
+	"postStartCommand": "pip install suite2p && mkdir -p example_data && s3fs djhub.vathes.datapub.elements:/workflow-calcium-imaging-data-v2 example_data -o multipart_size=530,endpoint=us-east-1,url=http://s3.amazonaws.com,public_bucket=1 && MYSQL_VER=8.0 docker compose down && MYSQL_VER=8.0 docker compose up --build --wait",
```
[the remark this answers is not in the document]
Comment (kabilar) [reply .devcontainer/devcontainer.json:19]: Discussed dependency conflict.  Will update in subsequent pull request.

File: .devcontainer/Dockerfile [hunk 7]
```diff
@@ -1,11 +1,24 @@
...
+    pip install --upgrade pip && \
+    pip install flake8 black pre-commit
+
+RUN \  
+    echo "Install CaImAn ----------" && \
+    pip install Pygments==2.14.0  && \
+    git clone https://github.com/datajoint/CaImAn.git  && \
+    pip install -r ./CaImAn/requirements.txt  && \
+    pip install ./CaImAn && \
+    rm -rf /tmp/CaImAn;
```
[the remark this answers is not in the document]
Comment (kabilar) [reply .devcontainer/Dockerfile:17]: Good idea.  Will add in subsequent pull request.

File: notebooks/tutorial.ipynb (hunk 1 above)
Comment (kabilar) [reply notebooks/tutorial.ipynb:64]: Discussed option to change to environment.  Will update in subsequent pull request.

File: notebooks/tutorial.ipynb (hunk 3 above)
Comment (kabilar) [reply notebooks/tutorial.ipynb]: Thanks!  I have incorporated your `drop_databases` function.

File: notebooks/tutorial.ipynb (hunk 2 above)
Comment (kabilar) [reply notebooks/tutorial.ipynb:280]: This was an issue with the data on S3.  It is now resolved.

**Pull request merged**